### PR TITLE
external sort testing

### PR DESCRIPTION
**Pull request by DeanAlvarez**
I refactored the external sorting algorithm so that I would be able to perform tests on it using python unittest

This also includes the test that I created. Since for the purposes of our dataset I only cared about user_id (I.E. if all the user_id are in order I don't care if the groups of one user_id are sorted by some other attribute or not) I simply put the user_id columns into two different lists which I compared.

### CODE REVIEW

File: tools/csv_sort.py [hunk 1]
```diff
@@ -41,60 +28,91 @@ def __lt__(self,other):
...
+# Algorithm info:
+# read and sort a chunk of the data using conventional methods
+# write sorted data to disk
+# repeat until all data is in sorted chunks
+# read small amount from each chunk
+# merge these mini-chunks
+# In particular, I merge these mini-chunks with a min-heap. I load the first element from each chunk into the heap
+# then I pop the top of the minheap and add another item from the chunk that the item which was popped origninated from
+# complexity: nlog(n), the sorting part is obviously nlog(n) and the merging part is more or less a heap-sort so it
+# is also nlog(n).
```
Comment (meritus) [inline tools/csv_sort.py]: NIT: Format these, usually we use an 80 char limit, well formatted.

File: tools/csv_sort.py [hunk 2]
```diff
@@ -25,8 +12,8 @@ class Row:
     def __init__(self,user_id,access_date,access_hour,access_minute,resource_attr_1,resource_attr_2,reader_number):
```
Comment (meritus) [inline tools/csv_sort.py]: 80 char limit.

Comment (meritus) [reply tools/csv_sort.py]: Also throughout the rest of file.

**Review by meritus — COMMENTED** [review]
Preliminary reviews.

File: tools/external_sort_test.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from csv_sort import external_sort
+import pandas as pd
+import os
+import unittest
+
+class TestExternalSortAccuracy(unittest.TestCase):
+    def test_accuracy(self):
```
Comment (tedim52) [inline tools/external_sort_test.py:8]: Think it would be good to add a comment underneath here to say exactly what this test case is testing.

**Review by tedim52 — APPROVED** [review]
LGTM.

File: tools/external_sort_test.py (hunk 3 above)
Comment (tedim52) [reply tools/external_sort_test.py:8]: Maybe something like #Test that user id column is sorted correctly

**Comment by DeanAlvarez** [comment]
I addressed all the comments and put all my code on under 80 lines

**Review by meritus — APPROVED** [review]

**Pull request merged**